TESTING.md

```diff
@@ -0,0 +1,274 @@
+The goal is that if the Component implementation changes, the component helpers should be the only
+code that needs to change to keep tests passing.
+
+Component helpers come in 3 flavors:
+
+- **Starter**: This helper types starts a Cypress chain. The return type is always
+  `Cypress.Chainable<JQuery>` and usually contains a `cy.get` and the selector usually contains
+  something that is unique to that component type. For example, the Modal helper uses
+  `[role=dialog]` with an option to disambiguate with a `data-testid`.
+- **Transform**: Imagine a web application is a collection of boxes that determine how components of
+  an application are laid out. A transform helper takes in a larger box and returns a smaller box,
+  further reducing the scope. Ideally these helpers contain no Cypress commands (which are
+  asynchronous), only jQuery calls (which are synchronous). Synchronous helper functions can be run
+  more than once at the discretion of the Cypress runtime. A good use-case for this type is to
+  replace CSS selectors
+- **Action**: Cypress comes with many baked in action commands such as `click` and `type`. A custom
+  action is useful to give a higher level semantic meaning to a series of low-level commands. For
+  example, a Select component might compose a Popup and a virtualized menu. This action may include
+  opening the select, finding the popup element, seeing if the desired item is currently visible,
+  scrolling the virtualized menu until the desired item is visible and clicking on the item. This
+  code would be repeated for every instance and has implementation knowledge of the component. A
+  high-level semantic action might be called "select" that wraps up this implementation detail to
+  free up automation and development engineers to think about the test case and not about
+  implementation details.
+
+### How to write a functional test
+
+A functional test describes the behavior and specifications of components. Functional tests should
+follow recommendations of behavior driven development where "specs" read like a specification and
+specs are broken down to a single assertion. Functional specifications should be readable by
+non-developers for verification. Most bugs occur when behavior is unspecified. UX issues are when
+specified behavior is not intuitive.
+
+All functional specifications start with a Storybook story. Stories serve both as a starting point
+for functional tests and visual tests. A specification is a starting point (a "given"), a series of
+actions on a component (a set of "when"s) and some type of expected outcome (a "then").
+
+For example:
+
+- Given the Default Modal story is rendered
+- When I click on the "Delete Item" button
+- Then the Modal should be open
+
+Put together, a sentence is formed: "Given the Default Modal story is rendered when I click on the
+"Delete Item" button then the Modal should be open"
```

Comment:
This is such a comprehensive and lengthy writeup I almost feel like we need a summary/conclusion paragraph. Your call though

Reply:
I added a "Summary" section. It is a bulleted section on what the different test types are. Let me know if this is what you were thinking or what other kind of wrap up we should do